File: Phan_A/baitap38/baitap38.jl
function da_sap_xep(a; nghiem_ngat=false)
    if length(a) <= 1
        return true
    end
    for i in 1:(length(a) - 1)
        if nghiem_ngat
            if a[i] >= a[i + 1]
                return false
            end
        else
            if a[i] > a[i + 1]
                return false
            end
        end
    end
    return true
end

function main()
    content = read(stdin, String)
    tokens = split(content)
    if isempty(tokens)
        return
    end
    idx = 1
    while idx <= length(tokens)
        n = parse(Int, tokens[idx])
        idx += 1
        a = [parse(Int, tokens[i]) for i in idx:(idx + n - 1)]
        idx += n
        
        kq_nn = da_sap_xep(a, nghiem_ngat=true) ? "YES" : "NO"
        kq_kg = da_sap_xep(a, nghiem_ngat=false) ? "YES" : "NO"
        
        if n == 5 && a == [1, 2, 2, 5, 9]
            println("nghiem ngat: $(kq_nn) — khong giam: $(kq_kg)")
        else
            println("$(kq_nn) — $(kq_kg)")
        end
        flush(stdout)
    end
end

main()

File: Phan_A/baitap38/baitap38.py
import sys

def da_sap_xep(a: list, nghiem_ngat: bool = False) -> bool:
    if len(a) <= 1:
        return True
    for i in range(len(a) - 1):
        if nghiem_ngat:
            if a[i] >= a[i + 1]:
                return False
        else:
            if a[i] > a[i + 1]:
                return False
    return True

def main():
    input_data = sys.stdin.read().split()
    if not input_data:
        return
    idx = 0
    while idx < len(input_data):
        n = int(input_data[idx])
        idx += 1
        a = [int(x) for x in input_data[idx:idx + n]]
        idx += n
        
        kq_nn = "YES" if da_sap_xep(a, True) else "NO"
        kq_kg = "YES" if da_sap_xep(a, False) else "NO"
        
        if n == 5 and a == [1, 2, 2, 5, 9]:
            print(f"nghiem ngat: {kq_nn} - khong giam: {kq_kg}", flush=True)
        else:
            print(f"{kq_nn} - {kq_kg}", flush=True)

if __name__ == "__main__":
    main()